polkadot-protocol/introduction/glossary.md

```diff
@@ -0,0 +1,384 @@
+Events in FRAME are implemented as a set of transient storage items that are inspected immediately after a block has executed and reset during block-initialization.
+
+## Executor
+
+A means of executing a function call in a given [runtime](#runtime) with a set of dependencies.
+There are two orchestration engines in Substrate, _WebAssembly_ and _native_.
+
+- The _native executor_ uses a natively compiled runtime embedded in the [node](#node) to execute calls.
+  This is a performance optimization that up-to-date nodes can take advantage of.
+
+- The _WebAssembly executor_ uses a [Wasm](#webassembly-wasm) binary and a Wasm interpreter to execute calls.
+  The binary is guaranteed to be up-to-date regardless of the version of the blockchain [node](#node) because it is persisted in the [state](#state) of the Substrate-based chain.
+
+## Extrinsic
+
+An extrinsic is a general term for a piece of data that is originated outside of the runtime, included into a block and leads to some action. This includes user-initiated transactions as well as inherents which are placed into the block by the block-builder.
+
+It is a SCALE encoded array typically consisting of a version number, signature, and varying data types indicating the resulting runtime function to be called. Extrinsics take two forms, "[inherents](#inherent-transactions)" and "[transactions](#transaction)". For more technical details see the [Polkadot spec](https://spec.polkadot.network/#id-extrinsics).
+
+## Existential Deposit
+
+The minimum balance an account is allowed to have in the [Balances pallet](https://paritytech.github.io/polkadot-sdk/master/pallet_balances/index.html).
+Accounts cannot be created with a balance less than the existential deposit amount.
+If an account balance drops below this amount, the Balances pallet uses [a FRAME System API](https://paritytech.github.io/substrate/master/frame_system/pallet/struct.Pallet.html#method.dec_ref) to drop its references to that account.
+If all of the references to an account are dropped, the account can be [reaped](https://paritytech.github.io/substrate/master/frame_system/pallet/struct.Pallet.html#method.allow_death).
+
+## Fork Choice Rule/Strategy
+
+A fork choice rule (or strategy) helps determine which chain is valid when reconciling amongst several forks within a network.  A common fork choice rule is the [longest chain](https://paritytech.github.io/polkadot-sdk/master/sc_consensus/struct.LongestChain.html), which the chain with the highest number of blocks gets selected.
+
+## FRAME
+
+An acronym for the _Framework for Runtime Aggregation of Modularized Entities_ that enables developers to create blockchain [runtime](#runtime) environments from a modular set of components called [pallets](#pallet).  It utilizes a set of procedural macros to construct runtimes.
+
+[Visit the Polkadot SDK docs For more detail on FRAME.](https://paritytech.github.io/polkadot-sdk/master/polkadot_sdk_docs/polkadot_sdk/frame_runtime/index.html)
```

Comment:
To prevent this flag, the first time you use an acronym, format it this way: 

Spell Out My Words (SOMW) 

This is the pattern the rule looks for the first time it sees an acronym.